File: gym-management-system-backend/src/main/java/com/gym/modules/exercise/domain/entity/Exercise.java
package com.gym.modules.exercise.domain.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.util.Date;

@Data
@TableName("gym_exercise")
public class Exercise {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String targetMuscle;
    private String equipment;
    private String difficulty;
    private String imageUrl;
    private String videoUrl;
    private String primaryMuscle;
    private String secondaryMuscles;
    private String movementPattern;
    private String steps;
    private String commonMistakes;
    private String alternatives;
    private String mediaLicense;
    private String tips;
    private String status;

    @TableField(fill = FieldFill.INSERT)
    private Date createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Date updateTime;

    @TableLogic
    private Integer deleted;
}

File: gym-management-system-frontend/src/api/exercise.ts
import request from '@/utils/request'

export interface Exercise {
  id?: number
  name: string
  targetMuscle?: string
  equipment?: string
  difficulty?: string
  imageUrl?: string
  videoUrl?: string
  steps?: string
  commonMistakes?: string
  tips?: string
  status?: string
}

export function getExerciseList(params: any) {
  return request({
    url: '/exercise/list',
    method: 'get',
    params
  })
}

export function getExerciseDetail(id: number) {
  return request({
    url: `/exercise/${id}`,
    method: 'get'
  })
}

export function getExerciseAlternatives(id: number) {
  return request({
    url: `/exercise/${id}/alternatives`,
    method: 'get'
  })
}

export function addExercise(data: Exercise) {
  return request({
    url: '/admin/exercise',
    method: 'post',
    data
  })
}

export function updateExercise(id: number, data: Exercise) {
  return request({
    url: `/admin/exercise/${id}`,
    method: 'put',
    data
  })
}

export function deleteExercise(id: number) {
  return request({
    url: `/admin/exercise/${id}`,
    method: 'delete'
  })
}
